tcp/clientconn.go

```diff
@@ -241,7 +240,11 @@ func (cc *ClientConn) Session() *Session {
 
 // Close closes connection without wait of ends Run function.
 func (cc *ClientConn) Close() error {
-	return cc.session.Close()
+	err := cc.session.Close()
+	if coapNet.IsCancelOrCloseError(err) {
```

Comment:
Also I don't think we need to check for Cancel or EOF here, so `errors.Is(err, net.ErrClosed)` should be enough here (and in `udp.Close`).